internal/testutil/acc/encryption_at_rest.go

```diff
@@ -47,33 +43,139 @@ func ConfigEARAzureKeyVault(projectID string, azure *admin.AzureKeyVault, useReq
 	return config
 }
 
-func ConfigAwsKms(projectID string, aws *admin.AWSKMSConfiguration, useDatasource, useRequirePrivateNetworking, useEnabledForSearchNodes bool) string {
+func ConfigAwsKmsWithRole(projectID, awsIAMRoleName, awsIAMRolePolicyName string, awsKms *admin.AWSKMSConfiguration, useDatasource, useRequirePrivateNetworking, useEnabledForSearchNodes bool) string {
 	requirePrivateNetworkingStr := ""
 	if useRequirePrivateNetworking {
-		requirePrivateNetworkingStr = fmt.Sprintf("require_private_networking = %t", aws.GetRequirePrivateNetworking())
+		requirePrivateNetworkingStr = fmt.Sprintf("require_private_networking = %t", awsKms.GetRequirePrivateNetworking())
 	}
-	enabledForSearchNodes := ""
+	enabledForSearchNodesStr := ""
 	if useEnabledForSearchNodes {
-		enabledForSearchNodes = fmt.Sprintf("enabled_for_search_nodes = %t", useEnabledForSearchNodes)
+		enabledForSearchNodesStr = fmt.Sprintf("enabled_for_search_nodes = %t", useEnabledForSearchNodes)
+	}
+	datasourceStr := ""
+	if useDatasource {
+		datasourceStr = EARDatasourceConfig()
 	}
 	config := fmt.Sprintf(`
+		%[2]s
+
+		resource "mongodbatlas_cloud_provider_access_setup" "setup_only" {
+			project_id    = %[1]q
+			provider_name = "AWS"
+		}
+	  
+		resource "mongodbatlas_cloud_provider_access_authorization" "auth_role" {
+			project_id = %[1]q
```

Comment:
```suggestion
			project_id = mongodbatlas_cloud_provider_access_setup.setup_only.project_id
```

this is ok, an alternative is to set the project_id in one resource, and refer from the others to that resource

Reply:
Yes, avoiding creating a dependency.